pyflagser/flagio.py

```diff
@@ -1,11 +1,73 @@
 """Implementation of input/output functions for .flag files"""
 
+import warnings
 import numpy as np
 import scipy.sparse as sp
 
+from ._utils import _extract_unweighted_graph, _extract_weighted_graph
 
-def loadflag(fname, fmt='csr', dtype=None):
-    """Load a ``.flag`` file, and return a matrix representation.
+
+def load_unweighted_flag(fname, fmt='csr', dtype=np.bool):
+    """Load a ``.flag`` file and return the adjacency matrix of the
+    directed/undirected unweighted graph it describes.
+
+    Parameters
+    ----------
+    fname : file, str, or pathlib.Path, required
+        Filename of extension ``.flag`` containing the information of a flag
+        matrix.
+
+    fmt : {'dense', 'dia', 'csr', 'csc', 'lil', ...}, optional,
+        default: ``'csr'``
+        Matrix format of the result. By default, a CSR sparse matrix is
+        returned. Keep in mind that some matrix formats do not track zero
+        values.
+
+    dtype : data-type, optional, default: ``np.bool``
+        Data-type of the resulting array.
+
+    Returns
+    -------
+    adjacency_matrix : matrix of shape (n_vertices, n_vertices) and format
+        `fmt`
+        Adjacency matrix of a directed/undirected unweighted graph. It is
+        understood as a boolean matrix. Off-diagonal, ``0`` or ``False`` values
+        denote abstent edges while non-``0`` or ``True`` values denote edges
+        which are present. Diagonal values are ignored.
+
+    Notes
+    -----
+    The input graphs cannot contain self-loops, i.e. edges that start and end
+    in the same vertex, therefore diagonal elements of the input adjacency
+    matrix will be ignored.
+
+    References
+    ----------
+    .. [1] D. Luetgehetmann, "Documentation of the C++ flagser library";
+           `GitHub: <https://github.com/luetge/flagser/blob/master/docs/\
+           documentation_flagser.pdf>`_.
+
+    """
+    with open(fname, 'r') as f:
+        next(f)
+        line = f.readline().strip()
+        vertices = list(map(dtype, line.split(' ')))
```

Comment:
Isn't it sufficient to just write this and the next two lines as
```
n_vertices = len(line.split(' '))
adjacency_matrix = sp.csr_matrix((n_vertices, n_vertices), dtype=dtype)
```
because we don't really care about the values of the vertices?